server.js

```
@@ -144,7 +146,8 @@ app.post('/apply', validate(), rateLimit(), function (req, res) {
       uri: req.originUri + '/images/' + filename
     });
 
-    renameJobs.push(async.apply(fs.rename, tmpPath, dest));
+    mv(tmpPath, dest, function(err) {
```

This would need to be rewritten to use `async.apply()` and pushed into `renameJobs`, as on the removed line. This ensures that all of the files (keep in mind that this could be used for lots of other types of Slack teams, and maybe they'll want to require multiple file uploads) get moved in parallel asynchronously. Also it makes sure that if any of the file operations produce an error, an error message can be shown to the user. As you have written it, any errors would be swallowed and the app would report a false success.

If you look at the `async` code starting on line 153, this is where the actual file operations take place, with a callback to handle the result (check for error or continue processing, etc), that should give you a better idea of what's going on.